### ISSUE TITLE
Cannot reproduce en<->fr BLEU scores with LSTM model

### ISSUE THREAD

**Issue by komeme** [facebookresearch/UnsupervisedMT#56]
Thank you for sharing your codes.

I successfully reproduced the BLEU scores on en<->fr translation with Transformer model by running your command written in `README.md`. 
However, I could not reproduce the BLEU scores with LSTM models which are reported on your paper(https://arxiv.org/abs/1804.07755).
Please tell me your training command to reproduce the BLEU scores reported on your paper on en<->fr translation using LSTM model.
Also, I want to know how long did it take to converge.

## details
the result reported on the paper is that
* en->fr 24.5
* fr->en 23.7

my result was that
* en->fr 11.18 (after 96epoch, 8days)
* fr->en 11.30(after 96epoch, 8days)

my settings
* 32 CPU
* 1 GPU (Tesla K80, 12GB)
* pytorch version 0.41
* python version 3.6.5

my command is here
```
python main.py --exp_name test3 --transformer False --n_enc_layers 4 --n_dec_layers 4 --share_enc 3 --share_dec 3 --share_lang_emb True --share_output_emb True --langs 'en,fr' --n_mono -1 --mono_dataset 'en:./data/mono/all.en.tok.60000.pth,,;fr:./data/mono/all.fr.tok.60000.pth,,' --para_dataset 'en-fr:,./data/para/dev/newstest2013-ref.XX.60000.pth,./data/para/dev/newstest2014-fren-src.XX.60000.pth' --mono_directions 'en,fr' --word_shuffle 3 --word_dropout 0.1 --word_blank 0.2 --pivo_directions 'fr-en-fr,en-fr-en' --pretrained_emb './data/mono/all.en-fr.60000.vec' --pretrained_out True --lambda_xe_mono '0:1,100000:0.1,300000:0' --lambda_xe_otfd 1 --otf_num_processes 30 --otf_sync_params_every 1000 --enc_optimizer adam,lr=0.0001 --epoch_size 500000 --stopping_criterion bleu_en_fr_valid,10 --batch_size 16
```

the full log file is here
https://github.com/komeme/public/files/2729329/train.log

**Comment by ijauregiCMCRC** [comment]
Hi,

I'm having the same problem.
I followed the paper's specifications and changed the following hyper-parameters when using LSTM cells for encoder and decoder:
- num of layers enc/dec=3
- word vec dim = 300
- hidden_dim =300

But I couldn't get more than 16 BLEU points in both directions. Do I need to change any other hyper-parameter?
Thanks,
